common/lib/sequential.ts

```diff
@@ -1,23 +1,39 @@
-const sequentialLocks = new Map< () => Promise< unknown >, Set< Promise< unknown > > >();
+type SequentialOptions = {
+	concurrent?: number;
+	max?: number;
+};
 
-// Ensures that calls to the provided function are executed sequentially
 export function sequential< Args extends unknown[], Return >(
-	fn: ( ...args: Args ) => Promise< Return >
+	fn: ( ...args: Args ) => Promise< Return >,
+	options?: SequentialOptions
 ) {
+	const concurrentCount = options?.concurrent ?? 1;
+	const maxQueueSize = options?.max;
+	const locks = new Set< Promise< unknown > >();
```

Comment:
```suggestion
	const locks = new Set< Promise< Return > >();
```
Good idea moving `locks` inside the closure 👍 It definitely simplifies things. We might as well use the correct return type now, though.